File: event-post-api/app/models/event.rb
class Event < ApplicationRecord
  has_many :comments, dependent: :destroy
  belongs_to :user
  has_one_attached :image, dependent: :destroy # `dependent: :destroy` を追加
  # その他のコード…
  has_many :likes, dependent: :destroy
  has_many :liked_users, through: :likes, source: :user
  validates :title, :date, :location, :description, presence: true
end

File: event-post-app/src/types/event.ts
import type { User } from '@/types/user';

export interface Event {
  id: number;
  title: string;
  date: string;
  location: string;
  description?: string;
  image_url?: string; // Optional, if you want to include an image URL
  createdAt: string;
  updatedAt: string;
  user: User;
}
